### ISSUE TITLE
pmf-pymc example is broken

### ISSUE THREAD

**Issue by fonnesbeck** [pymc-devs/pymc#830]
The referenced data for this example seems not to exist, and print statements are legacy Python.

**Comment by twiecki** [comment]
ping @macks22

**Comment by macks22** [comment]
I assume you're referring to the "jokes" directory? The [csv file](https://github.com/pymc-devs/pymc3/blob/master/pymc3/examples/data/jester-dataset-v1-dense-first-1000.csv), the [train/test split](https://github.com/pymc-devs/pymc3/tree/master/pymc3/examples/data/6bb8d06c69c0666e6da14c094d4320d115f1ffc8) and the [pmf-map-d5](https://github.com/pymc-devs/pymc3/tree/master/pymc3/examples/data/pmf-map-d5) directory are all present. As for "legacy" print statements, would you like them converted to Python 3.X function-style print statements? If so, will a v4 notebook be compatible or are only < v4 notebooks supported?

**Comment by twiecki** [comment]
@macks22 That's how I had remembered it. @fonnesbeck why weren't you able to run the files? Perhaps we aren't referencing the correct directory.

Changing to `print()` is 2 and 3 compatible so that one is an easy choice.

I think ipython nb v4 is fine.

**Comment by ColCarroll** [comment]
This is still broken -- looks like the files can be found here: http://eigentaste.berkeley.edu/dataset/

They are ~97kb.  I will add them back in a separate PR from #1444, and update print statements.

```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-6-46c88078b60d> in <module>()
      7 # Let's see for ourselves. Load the jokes.
      8 joke_dir = '../data/jokes'
----> 9 files = [os.path.join(joke_dir, fname) for fname in os.listdir(joke_dir)]
     10 jokes = [fname for fname in files if fname.endswith('txt')]
     11 nums = [filter(lambda c: c.isdigit(), fname) for fname in jokes]

FileNotFoundError: [Errno 2] No such file or directory: '../data/jokes'
```

**Comment by twiecki** [comment]
Can we download them on the fly if they don't exist?

**Comment by ColCarroll** [comment]
Looking closer, the files are used in one place, and only to access the
best and worst rated jokes.  I see three choices:
1. It could be removed, but I think it adds more character to the example
(if we're rating jokes, we should get to see the jokes)
2. It could be hard coded.  This would be sad for anyone who wanted to
interact with the notebook.
3. We download on the fly.  I can't figure out how to do that in a way that
is python 2/3 compatible, other than adding the `requests` library.  Also
there's some light post-processing (the files are available as html, but we
might want to display them as text).
4. We add 100 files to pymc3/examples/data/jokes, 1.txt through 100.txt.
I've already done this, and it runs as expected.
5. We combine all 100 into jokes.csv, with line number corresponding to
joke number.  This is a <50kb file.

I like choice 5 the best.

Another problem is that `pmf.draw_samples(5000)` appears to take hours to
run (didn't complete last night).  I'm going to lower the number of samples
to just make sure it runs, and maybe someone else can work on the
performance.

**Comment by twiecki** [comment]
Closed by #1447.
